src/components/TextField/TextArea.js

```diff
@@ -0,0 +1,279 @@
+
+    componentDidMount = () => {
+        autosize(this.inputRef);
+    }
+
+    handleMouseEnter = () => {
+        this.setState({ hasMouseOver: true });
+    }
+
+    handleMouseLeave = () => {
+        this.setState({ hasMouseOver: false });
+    }
+
+    handleInputRef = (ref) => {
+        const { componentRef } = this.props;
+        this.inputRef = ref;
+        if (componentRef) {
+            componentRef(ref);
+        }
+    }
+
+    handleInputBlur = (event) => {
+        const { onBlur } = this.props;
+        this.setState({
+            hasFocus: false,
+            showTooltip: false,
+        });
+        if (onBlur) {
+            onBlur(event);
+        }
+    }
+
+    handleInputChange = (event) => {
+        const { value } = this.inputRef;
+        const { onChange } = this.props;
+        if (typeof value !== 'undefined') {
+            this.setState({ value });
+        }
+        if (onChange) {
+            onChange(value, event);
+        }
+    }
+
+    handleInputFocus = (event) => {
+        const { onFocus } = this.props;
+        this.setState({
+            hasFocus: true,
+            showTooltip: true,
+        });
+        if (onFocus) {
+            onFocus(event);
+        }
+    }
+
+    handleInputKeyDown = (event) => {
+        const { onKeyDown } = this.props;
+        const { onEnterKey } = this.props;
+        if (onKeyDown) {
+            onKeyDown(event.key, event);
+        }
+        if (onEnterKey && event.key === 'Enter') {
+            onEnterKey(event);
+        }
+    }
+
+    handleInputKeyPress = (event) => {
+        const { onKeyPress } = this.props;
+        if (onKeyPress) {
+            onKeyPress(event.key, event);
+        }
+    }
+
+    handleInputKeyUp = (event) => {
+        const { onKeyUp } = this.props;
+        if (onKeyUp) {
+            onKeyUp(event.key, event);
+        }
+    }
+
+    /**
+     * wrapInputWithTooltip
+     * provide a string or component to tooltipError or tooltipHint and this
+     * wrapper will be used to create a tooltipBox (an error box is used for tooltipError)
+     * @param {Element} textarea - the textarea field
+     * @param {Component|string} tooltip - the error or hint tooltip
+     */
+    wrapInputWithTooltip = (textarea, tooltip) => {
+        const { DEFAULT, ERROR } = TooltipBoxStatus;
+        return (
+            tooltip ?
+                <Tooltip
+                    position={TooltipPosition.BOTTOM_CENTER}
+                    showTooltip={this.state.showTooltip}
+                    tooltip={
+                        <TooltipBox
+                            status={this.props.tooltipError ? ERROR : DEFAULT}
+                        >
+                            {tooltip}
+                        </TooltipBox>
+                    }
+                >
+                    {textarea}
+                </Tooltip> :
+                textarea
+        );
+    }
+
+    render() {
+        /* eslint-disable jsx-a11y/label-has-for */
```

Comment:
Should we move this line closer to the actual label code instead of wrapping the whole `render` function?

Reply:
Thanks for spotting this, originally label was inside the JSX so I couldn't place the comment alongside, then when I moved the label out of the JSX I didn't move the comment – good suggestion